test/common/network/filter_manager_impl_test.cc

```diff
@@ -73,13 +74,15 @@ TEST_F(NetworkFilterManagerTest, All) {
   MockWriteFilter* write_filter(new MockWriteFilter());
   MockFilter* filter(new LocalMockFilter());
 
-  FilterManagerImpl manager(connection_);
+  FilterManagerImpl manager(connection_, socket_);
   manager.addReadFilter(ReadFilterSharedPtr{read_filter});
   manager.addWriteFilter(WriteFilterSharedPtr{write_filter});
   manager.addFilter(FilterSharedPtr{filter});
 
   read_filter->callbacks_->upstreamHost(Upstream::HostDescriptionConstSharedPtr{host_description});
   EXPECT_EQ(read_filter->callbacks_->upstreamHost(), filter->callbacks_->upstreamHost());
+  EXPECT_EQ(&read_filter->callbacks_->socket(), &socket_);
+  EXPECT_EQ(&write_filter->write_callbacks_->socket(), &socket_);
```

Comment:
Should both of those point to the same (presumably downstream) socket, and not separate downstream and upstream sockets?

Reply:
Each connection object has its own filter manager so the ClientConnection subclass will have the downstream connection/socket, and the ServerConnection subclass will have the upstream connection/socket. Both upstream and downstream connections can have read and write filters.